pkg/image/qemu.go

```diff
@@ -1,16 +1,127 @@
+/*
+Copyright 2018 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package image
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
 	"os/exec"
 
+	"github.com/golang/glog"
 	"github.com/pkg/errors"
 )
 
+const (
+	networkTimeoutSecs = 3600    //max is 10000
+	maxMemory          = 1 << 30 //value from OpenStack Nova
+	maxCPUSecs         = 30      //value from OpenStack Nova
+)
+
+// ConvertQcow2ToRaw converts a local qcow2 image to raw format
 func ConvertQcow2ToRaw(src, dest string) error {
-	cmd := exec.Command("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
-	err := cmd.Run()
+	err := validate(src, "qcow2")
 	if err != nil {
-		return errors.Wrap(err, "could not convert qcow2 image to raw")
+		return errors.Wrap(err, "image validation failed")
 	}
+
+	_, err = execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
+	if err != nil {
+		return errors.Wrap(err, "could not convert local qcow2 image to raw")
+	}
+
 	return nil
 }
+
+// ConvertQcow2ToRawStream converts an http accessible qcow2 image to raw format without locally caching the qcow2 image
+func ConvertQcow2ToRawStream(url *url.URL, dest string) error {
```

Comment:
Should this be a functional test?

Reply:
I'd say so.  Will depend on @copejon 's server container in the environment.

Reply:
@aglitke this test actually starts an embedded http server that serves up image files.  but yeah may be redundant in functional test environment